app/models/post.rb

```diff
@@ -1,5 +1,20 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   has_many :comments
   has_many :likes
+
+  after_create :increment_user_posts_counter
+  after_destroy :decrement_user_posts_counter
+
+  def increment_user_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def decrement_user_posts_counter
+    author.decrement!(:posts_counter)
+  end
```

Comment:
- [OPTIONAL] Great work for using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html). There's a better way with lesser line of code.
- You can use `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.
-  It should look something similar to this. :point_down: 

```
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```